### PR TITLE
Bind both to IPv4 and IPv6 with special "any" value

### PR DESCRIPTION
**Pull request by kxepal**
This also removes need in special wrapper to handle atom as default
for config:get call.

COUCHDB-2783

### CODE REVIEW

**Comment by rnewson** [comment]
This only helps for the private 5986 port for 2.0 which should really only bind to localhost. Look at https://github.com/apache/couchdb-chttpd/blob/master/src/chttpd.erl#L112.

**Comment by kxepal** [comment]
@rnewson I know. See also https://github.com/apache/couchdb-chttpd/pull/60

**Comment by rnewson** [comment]
+1

**Comment by rnewson** [comment]
hrm, I retract the +1, sorry.

File: src/couch_httpd.erl [hunk 1]
```diff
@@ -1127,11 +1127,10 @@ check_for_last(#mp{buffer=Buffer, data_fun=DataFun}=Mp) ->
                 data_fun = DataFun2})
     end.
 
+validate_bind_address(any) -> any;
+validate_bind_address("any") -> any;
```
Comment (rnewson) [inline src/couch_httpd.erl]: I don't think the validate_bind_address function should work this way, it should _only_ validate. It now returns 'ok' or 'any' or throws, which is quite confusing. The intention is correct though, to allow config to hold "any" and have that be the atom any. Have another go :)

Comment (kxepal) [inline src/couch_httpd.erl]: Right (:

**Comment by kxepal** [comment]
@rnewson  Fixed that problem. Please take another look.

File: src/couch_httpd.erl [hunk 2]
```diff
@@ -1127,11 +1131,9 @@ check_for_last(#mp{buffer=Buffer, data_fun=DataFun}=Mp) ->
                 data_fun = DataFun2})
     end.
 
+validate_bind_address(any) -> ok;
 validate_bind_address(Address) ->
     case inet_parse:address(Address) of
-        {ok, _} -> ok;
```
Comment (rnewson) [inline src/couch_httpd.erl]: change this back to `ok`, you never use the return value anyway.

File: src/couch_httpd.erl [hunk 3]
```diff
@@ -88,7 +89,10 @@ start_link(https) ->
          {ssl_opts, SslOpts}],
     start_link(https, Options).
 start_link(Name, Options) ->
-    BindAddress = with_default(config:get("httpd", "bind_address"), any),
+    BindAddress = case config:get("httpd", "bind_address", "any") of
+                      "any" -> any;
+                      Else -> Else
+                  end,
     validate_bind_address(BindAddress),
```
Comment (rnewson) [inline src/couch_httpd.erl]: say `ok = ` here.

File: src/couch_httpd.erl (hunk 2 above)
Comment (kxepal) [inline src/couch_httpd.erl]: Accidental change. Thanks.

**Comment by rnewson** [comment]
+1, please squash before merging.

**Pull request merged**